pex/network_configuration.py

```diff
@@ -53,3 +56,7 @@ def create(
             cert=cert,
             client_cert=client_cert,
         )
+
+    def headers_as_dict(self):
+        # type: () -> Dict[str, str]
+        return dict(header.split(":", 1) for header in self.headers)
```

Comment:
AFAICT we don't validate the headers anywhere, and if one doesn't contain a `:` then this will fail in an unhelpful way. Whereas previously pip would have issued a helpful `Could not parse header` warning and continued. Might be worth validating the headers in `create()` so that users get an explicit, helpful message if their header string doesn't contain a `:`.

Reply:
Yup, good point. Fixed.